src/server/zset_family.cc

```diff
@@ -1058,6 +1058,36 @@ struct SetOpArgs {
   bool with_scores = false;
 };
 
+void HandleOpStatus(ConnectionContext* cntx, OpStatus op_status) {
+  switch (op_status) {
+    case OpStatus::INVALID_FLOAT:
+      return (*cntx)->SendError("weight value is not a float", kSyntaxErrType);
+    default:
+      return (*cntx)->SendError(op_status);
+  }
+}
+
+void mergeScoredMaps(ConnectionContext* cntx, const std::vector<OpResult<ScoredMap>>& maps,
+                     const AggType agg_type, ScoredMap& result) {
+  for (auto op_res : maps) {
+    if (op_res.status() == OpStatus::SKIPPED)
+      continue;
+
+    if (!op_res)
+      return (*cntx)->SendError(op_res.status());
```

Comment:
that's not good. in case of an error, you send the error here but continue processing in the caller where the original code stopped returned from the caller. This is why I suggested returning `OpResult<ScoredMap>` so that the flow control would stay at the caller side.

Reply:
fix: https://github.com/dragonflydb/dragonfly/pull/2164/commits/18b083e001bbbdf64e1288afd443eee2717c32c9